src/unix/linux_like/android/b64/x86_64/mod.rs

```diff
@@ -41,6 +42,46 @@ s! {
         pub st_ctime_nsec: ::c_long,
         __unused: [::c_long; 3],
     }
+
+    pub struct _libc_fpxreg {
+        pub significand: [u16; 4],
+        pub exponent: u16,
+        __padding: [u16; 3], // TODO: ignore padding
+    }
+
+    pub struct _libc_xmmreg {
+        pub element: [u32; 4],
+    }
+
+    pub struct _libc_fpstate {
+        pub cwd: u16,
+        pub swd: u16,
+        pub ftw: u16,
+        pub fop: u16,
+        pub rip: u64,
+        pub rdp: u64,
+        pub mxcsr: u32,
+        pub mxcr_mask: u32,
+        pub _st: [_libc_fpxreg; 8],
+        pub _xmm: [_libc_xmmreg; 16],
+        __private: [u32; 24], // TODO: ignore padding
+    }
+
+    pub struct mcontext_t {
+        pub gregs: [greg_t; 23],
+        pub fpregs: *mut _libc_fpstate,
+        __private: [u64; 8], // TODO: ignore padding
+    }
+
+    pub struct ucontext_t {
+        pub uc_flags: ::c_ulong,
+        pub uc_link: *mut ucontext_t,
+        pub uc_stack: ::stack_t,
+        pub uc_mcontext: mcontext_t,
+        // FIXME this is actually a union
+        pub uc_sigmask64: ::sigset_t,
```

Comment:
Should use an `union`.

Reply:
Same here https://github.com/rust-lang/libc/pull/1622#discussion_r361042076